File: CiaiControllerSDKForJava/src/main/java/com/ciai/controller/sdk/model/Result.java
package com.ciai.controller.sdk.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * 统一响应结果
 */
public class Result<T> {

    @JsonProperty("code")
    private String code;

    @JsonProperty("message")
    private String message;

    @JsonProperty("data")
    private T data;

    public Result() {
    }

    public Result(String code, String message) {
        this.code = code;
        this.message = message;
    }

    public Result(String code, String message, T data) {
        this.code = code;
        this.message = message;
        this.data = data;
    }

    @JsonIgnore
    public boolean isSuccess() {
        return CommonCode.SUCCESS.equals(code);
    }

    // Getters and Setters
    public String getCode() {
        return code;
    }

    public void setCode(String code) {
        this.code = code;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public T getData() {
        return data;
    }

    public void setData(T data) {
        this.data = data;
    }

    // Static factory methods
    public static <T> Result<T> success() {
        return new Result<>(CommonCode.SUCCESS, "Success");
    }

    public static <T> Result<T> success(T data) {
        return new Result<>(CommonCode.SUCCESS, "Success", data);
    }

    public static <T> Result<T> success(String message, T data) {
        return new Result<>(CommonCode.SUCCESS, message, data);
    }

    public static <T> Result<T> failed() {
        return new Result<>(CommonCode.FAILED, "Failed");
    }

    public static <T> Result<T> failed(String message) {
        return new Result<>(CommonCode.FAILED, message);
    }

    public static <T> Result<T> failed(String code, String message) {
        return new Result<>(code, message);
    }

    public static <T> Result<T> unauthorized() {
        return new Result<>(CommonCode.UNAUTHORIZED, "Unauthorized");
    }

    public static <T> Result<T> timeout() {
        return new Result<>(CommonCode.TIMEOUT, "Timeout");
    }

    public static <T> Result<T> serverError() {
        return new Result<>(CommonCode.SERVER_ERROR, "Server Error");
    }

    public static <T> Result<T> serverError(String message) {
        return new Result<>(CommonCode.SERVER_ERROR, message);
    }

    public static <T> Result<T> parametersMissing() {
        return new Result<>(CommonCode.PARAMETERS_MISSING, "Parameters Missing");
    }

    public static <T> Result<T> parametersMissing(String message) {
        return new Result<>(CommonCode.PARAMETERS_MISSING, message);
    }
}

File: CiaiControllerSDK/Models/Result.cs
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace CiaiControllerSDK.Models
{
    /// <summary>
    /// 统一响应结果
    /// </summary>
    public class Result<T>
    {
        /// <summary>
        /// 状态码
        /// </summary>
        [JsonPropertyName("code")]
        public string Code { get; set; }

        /// <summary>
        /// 消息
        /// </summary>
        [JsonPropertyName("message")]
        public string Message { get; set; }

        /// <summary>
        /// 数据
        /// </summary>
        [JsonPropertyName("data")]
        public T Data { get; set; }

        /// <summary>
        /// 是否成功
        /// </summary>
        [JsonIgnore]
        public bool IsSuccess => Code == CommonCode.Success;

        public static Result<T> Success()
        {
            return new Result<T>
            {
                Code = CommonCode.Success,
                Message = "Success"
            };
        }

        public static Result<T> Success(T data)
        {
            return new Result<T>
            {
                Code = CommonCode.Success,
                Message = "Success",
                Data = data
            };
        }

        public static Result<T> Success(string message, T data)
        {
            return new Result<T>
            {
                Code = CommonCode.Success,
                Message = message,
                Data = data
            };
        }

        public static Result<T> Failed()
        {
            return new Result<T>
            {
                Code = CommonCode.Failed,
                Message = "Failed"
            };
        }

        public static Result<T> Failed(string message)
        {
            return new Result<T>
            {
                Code = CommonCode.Failed,
                Message = message
            };
        }

        public static Result<T> Failed(string code, string message)
        {
            return new Result<T>
            {
                Code = code,
                Message = message
            };
        }

        public static Result<T> Unauthorized()
        {
            return Failed(CommonCode.Unauthorized, "Unauthorized");
        }

        public static Result<T> Timeout()
        {
            return Failed(CommonCode.Timeout, "Timeout");
        }

        public static Result<T> ServerError(string message = "Server Error")
        {
            return Failed(CommonCode.ServerError, message);
        }

        public static Result<T> ParametersMissing(string message = "Parameters Missing")
        {
            return Failed(CommonCode.ParametersMissing, message);
        }
    }

    /// <summary>
    /// 通用状态码
    /// </summary>
    public static class CommonCode
    {
        public const string Success = "message.common.success";
        public const string Failed = "message.common.failed";
        public const string Unauthorized = "message.common.unauthorized";
        public const string Timeout = "message.common.timeout";
        public const string ServerError = "message.common.server.error";
        public const string ParametersMissing = "message.common.parameters.missing";
    }
}
